sw/device/silicon_creator/lib/drivers/watchdog.c

```diff
@@ -0,0 +1,46 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/watchdog.h"
+
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+
+#include "aon_timer_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pwrmgr_regs.h"
+
+enum {
+  kBase = TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR,
+  kPwrMgrBase = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
+};
+
+#define AON_TIMER_RATE 200000
+
+void watchdog_init(uint32_t timeout_ms) {
+  // Tell pwrmgr we want watchdog reset events to reset the chip.
+  abs_mmio_write32(kPwrMgrBase + PWRMGR_RESET_EN_REG_OFFSET,
+                   bitfield_bit32_write(0, PWRMGR_RESET_EN_EN_1_BIT, true));
+  abs_mmio_write32(kPwrMgrBase + PWRMGR_CFG_CDC_SYNC_REG_OFFSET, 1);
```

Comment:
Should these writes to pwrmgr registers occur after the timer is configured? Would it make sense to move them to the if statement?

Reply:
No, this is more like global configuration: we want pwrmgr to recognize the watchdog as a reset source regardless of whether the watchdog is actually enabled.